test/rdoc/test_rdoc_ri_driver.rb

```diff
@@ -32,8 +32,8 @@ def setup
   end
 
   def teardown
-    ENV['RI'] = @orig_ri
-    FileUtils.rm_rf @tmpdir
+    defined?(@orig_ri) and ENV['RI'] = @orig_ri
```

Comment:
when my tests were failing, the teardown here was failing as well, hence the guard.

Reply:
Where your tests failed, `mkdir_p`?

Reply:
@nobu I don't remember nor I simply reproduce. However I still think keeping that is a good idea, if you ever fail at some point in the setup, you don't want to fail as well here.

Moreover the `RDoc::TestCase` class already uses that pattern:

```
  ##
  # Abstract test-case teardown

  def teardown
    ENV["HOME"] = @orig_home if defined?(@orig_home)

    super
  end
```

If you find that it is superfluous I can still remove this bit.

Reply:
I see, just curious.